File: 3213-count-subarrays-where-max-element-appears-at-least-k-times/count-subarrays-where-max-element-appears-at-least-k-times.cpp
class Solution {
public:
    long long countSubarrays(vector<int>& nums, int k) {
        int n = nums.size();
        int max_num = *max_element(nums.begin(), nums.end());
        long long ans = 0 ; 
        int count = 0; 
        int left = 0; 

        for (int right = 0; right<n; right++){
            if (nums[right] == max_num){
                count++;
            }
            while(count >= k){
                ans += (n-right);
                if (nums[left] == max_num){
                    count--;
                }
                left++;
            }
        }
        return ans;
        
    }
};

File: 3213-count-subarrays-where-max-element-appears-at-least-k-times/count-subarrays-where-max-element-appears-at-least-k-times.py
class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
          max_ele = max(nums)
          n = len(nums)
          ans = 0 
          cnt = 0 
          left = 0 
          for right in range(n):
            if nums[right] == max_ele: 
               cnt += 1 
            
            while cnt >= k:
                ans += n - right  
                if nums[left] == max_ele:
                    cnt -= 1
                left += 1
          
          return ans  ## sornam.t